protocol/messenger_linkpreview.go

```diff
@@ -116,23 +145,54 @@ func GetURLs(text string) []string {
 		// case-sensitive, some websites encode base64 in the path, etc.
 		parsedURL.Host = strings.ToLower(parsedURL.Host)
 
-		idx := parsedURL.String()
-		// Removes the spurious trailing forward slash.
-		idx = strings.TrimRight(idx, "/")
-		if _, exists := indexed[idx]; exists {
+		url := parsedURL.String()
+		url = strings.TrimRight(url, "/") // Removes the spurious trailing forward slash.
+		if _, exists := indexedUrls[url]; exists {
 			continue
+		}
+
+		metadata := URLUnfurlingMetadata{
+			URL:               url,
+			IsStatusSharedURL: IsStatusSharedURL(url),
+		}
+
+		if !URLUnfurlingSupported(rawURL) {
+			metadata.Permission = URLUnfurlingNotSupported
+		} else if metadata.IsStatusSharedURL {
+			metadata.Permission = URLUnfurlingAllowed
 		} else {
-			indexed[idx] = nil
-			urls = append(urls, idx)
+			switch s.URLUnfurlingMode {
+			case settings.URLUnfurlingAlwaysAsk:
+				metadata.Permission = URLUnfurlingAskUser
+			case settings.URLUnfurlingEnableAll:
+				metadata.Permission = URLUnfurlingAllowed
+			case settings.URLUnfurlingDisableAll:
+				metadata.Permission = URLUnfurlingForbiddenBySettings
+			default:
+				metadata.Permission = URLUnfurlingForbiddenBySettings
+			}
 		}
 
-		// This is a temporary limitation solution,
-		// should be changed with https://github.com/status-im/status-go/issues/4235
-		if len(urls) == UnfurledLinksPerMessageLimit {
+		result.URLs = append(result.URLs, metadata)
+	}
+
+	return result
+}
+
+// Deprecated: GetURLs is deprecated in favor of more generic GetTextURLsToUnfurl.
```

Comment:
The deprecation warning should ideally be in the public API too, or just there.

Speaking about this @igor-sirotin, if the new endpoint doesn't require too many changes on mobile, I can tell we will soon be able to remove the deprecated endpoint. Thanks for evolving status-go in a backwards compatible way!

Reply:
Will add it there too 👍

Reply:
The new endpoint doesn't require much to do, but there's something to code.
dm me when you start working on this, I'll let you know of desktop solution with this endpoint 🙂